Calcify/Eggsamples.lean

```diff
@@ -0,0 +1,14 @@
+import Calcify.Basic
+
+/--
+info: Try this: calc
+    x
+    _ = x * x * (x * (x * x)) := (h x (x * x))
+    _ = x * x * x := congrArg (fun a'2521 => x * x * a'2521) (Eq.symm (h x x))
```

Comment:
Much better, @goens, right?

Reply:
yeah, that's really cool!

Reply:
Although I suggest that egg itself should produce less redundant proof objects, lest it becomes a bottle neck for large proofs during type checking or in terms of oleans files. Maybe the `simplify` that I am using here could become a general tool (maybe without floating `trans` out)